File: monorepo_go/packages/shared/shared.go
package shared

import "fmt"

type User struct {
	ID   int
	Name string
}

type Role string

const (
	Admin Role = "admin"
	User_ Role = "user"   // trailing underscore to avoid colliding with type
	Guest Role = "guest"
)

var DefaultRole = User_

func FormatUser(u User) string {
	return fmt.Sprintf("%d:%s", u.ID, u.Name)
}

File: monorepo_kotlin/packages/shared/src/main/kotlin/mono/shared/Shared.kt
package mono.shared

data class User(val id: Int, val name: String)

enum class Role { Admin, User, Guest }

val DEFAULT_ROLE = Role.User

object Util {
    fun formatUser(u: User): String = "${u.id}:${u.name}"
    fun hello(msg: String): String = "hello, $msg"
}

// Top-level extension function — Kotlin-specific shape.
fun String.quoted(): String = "\"$this\""
